pandas/core/groupby/groupby.py

```diff
@@ -2489,7 +2553,8 @@ def cumprod(self, axis=0, *args, **kwargs):
         """
         nv.validate_groupby_func("cumprod", args, kwargs, ["numeric_only", "skipna"])
         if axis != 0:
-            return self.apply(lambda x: x.cumprod(axis=axis, **kwargs))
+            f = lambda x: x.cumprod(axis=axis, **kwargs)
+            return self._python_apply_general(f, self._selected_obj, is_transform=True)
```

Comment:
can we not just call `.tranform()`?

Reply:
Not currently. `groupby(...).transform` uses `apply` on each group when testing the slow path (even if fast path works). `apply` will work column by column. This makes it so that `apply` with e.g. `lambda x: foo(x, axis=1)` fails since `x` will be a Series.